### PR TITLE
Create biz licence partial save endpoint

### PR DESCRIPTION
**Pull request by ruaraya**
Create biz licence partial save endpoint. The modifications include: 

- Creation of new service for biz licence application partial save
-  Implementation of related unit tests and integrated tests
- Replace mock mapper with concrete mapper in permit application manager test
- General refactoring and formatting 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

### CODE REVIEW

File: src/Spd.Resource.Repository/SharedRepositoryFuncs.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.Dynamics.CRM;
+using Spd.Resource.Repository.LicenceApplication;
+using Spd.Utilities.Dynamics;
+
+namespace Spd.Resource.Repository;
+internal static class SharedRepositoryFuncs
+{
+    public static void ProcessCategories(this DynamicsContext _context, IEnumerable<WorkerCategoryTypeEnum> categories, spd_application app)
```
Comment (peggy-quartech) [inline src/Spd.Resource.Repository/SharedRepositoryFuncs.cs:8]: if we think these functions are common shared function. We should not use extension method. So, context will just be an input parameter. Do not make it a context extension. Context extension methods should be in Utitliy Dynamics layer.

File: src/Spd.Manager.Licence/Mappings.cs [hunk 2]
```diff
@@ -179,6 +182,26 @@ public Mappings()
...
+            .ForMember(d => d.ManagerSurname, opt => opt.MapFrom(s => s.BizManagerContactInfo.Surname))
+            .ForMember(d => d.ManagerMiddleName1, opt => opt.MapFrom(s => s.BizManagerContactInfo.MiddleName1))
+            .ForMember(d => d.ManagerMiddleName2, opt => opt.MapFrom(s => s.BizManagerContactInfo.MiddleName2))
+            .ForMember(d => d.ManagerEmailAddress, opt => opt.MapFrom(s => s.BizManagerContactInfo.EmailAddress))
+            .ForMember(d => d.ManagerPhoneNumber, opt => opt.MapFrom(s => s.BizManagerContactInfo.PhoneNumber))
+            .ForPath(d => d.PrivateInvestigatorSwlInfo.ContactId, opt => opt.MapFrom(s => s.PrivateInvestigatorSwlInfo == null ? null : s.PrivateInvestigatorSwlInfo.ContactId))
+            .ForPath(d => d.PrivateInvestigatorSwlInfo.LicenceId, opt => opt.MapFrom(s => s.PrivateInvestigatorSwlInfo == null ? null : s.PrivateInvestigatorSwlInfo.LicenceId))
+            .ForPath(d => d.PrivateInvestigatorSwlInfo.BizContactId, opt => opt.MapFrom(s => s.PrivateInvestigatorSwlInfo == null ? null : s.PrivateInvestigatorSwlInfo.BizContactId));
```
Comment (peggy-quartech) [inline src/Spd.Manager.Licence/Mappings.cs:203]: PrivateInvestigatorSwlInfo is not saved in BizContact, so, there should be no bizContactId

File: src/Spd.Manager.Licence/LicenceAppDocumentManager.cs [hunk 3]
```diff
@@ -15,28 +16,35 @@ internal partial class LicenceAppDocumentManager :
...
         DocumentTypeEnum? docType1 = Mappings.GetDocumentType1Enum(command.Request.LicenceDocumentTypeCode);
         DocumentTypeEnum? docType2 = Mappings.GetDocumentType2Enum(command.Request.LicenceDocumentTypeCode);
 
         LicenceApplicationResp app = await _licenceAppRepository.GetLicenceApplicationAsync(command.AppId, cancellationToken);
         if (app == null)
             throw new ArgumentException("Invalid application Id");
 
+        if (app.WorkerLicenceTypeCode == WorkerLicenceTypeEnum.SecurityBusinessLicence)
+            bizLicApplicationResp = await _bizLicApplicationRepository.GetBizLicApplicationAsync(command.AppId, cancellationToken);
+
```
[the remark this answers is not in the document]
Comment (ruaraya) [reply src/Spd.Manager.Licence/LicenceAppDocumentManager.cs:52]: Fixed

File: src/Spd.Manager.Licence/Mappings.cs (hunk 2 above)
Comment (ruaraya) [reply src/Spd.Manager.Licence/Mappings.cs]: Fixed

File: src/Spd.Resource.Repository/SharedRepositoryFuncs.cs (hunk 1 above)
Comment (ruaraya) [reply src/Spd.Resource.Repository/SharedRepositoryFuncs.cs]: Fixed

**Review by peggy-quartech — APPROVED** [review]

**Pull request merged**